olp-cpp-sdk-authentication/src/AuthenticationClient.cpp

```diff
@@ -258,34 +228,40 @@ AuthenticationClient::Impl::Impl(const std::string& authentication_server_url,
           std::make_shared<
               Atomic<utils::LruCache<std::string, SignInUserResult> > >(
               token_cache_limit)),
-      network_config_() {}
+      network_settings_() {}
 
 client::CancellationToken AuthenticationClient::Impl::SignInClient(
     const AuthenticationCredentials& credentials,
     const chrono::seconds& expires_in,
     const AuthenticationClient::SignInClientCallback& callback) {
+  if (!network_) {
+    AuthenticationError result({static_cast<int>(http::ErrorCode::IO_ERROR),
+                                "Cannot sign in while offline"});
+    callback(result);
```

Comment:
Is this a sync or an async callback?

Reply:
same approach as in your reference: https://github.com/heremaps/here-olp-edge-sdk-cpp/pull/96/files#diff-bbc42a63ded02006254f767d3dc2cd3bR72

Reply:
Except that one is in the context of the async TaskScheduler. I am not sure in which context this is happening.

Reply:
done